### PR TITLE
Fix HamiltonianMC init to forward arguments

### PR DESCRIPTION
**Pull request by ColCarroll**
I think @fonnesbeck already made these changes to `NUTS`, I am just following along. This lets you pass arguments upstream to `BaseHMC` from `HamiltonianMC`. 

I am also updating the default `target_accept` for HMC.

### CODE REVIEW

File: pymc3/step_methods/hmc/hmc.py [hunk 1]
```diff
@@ -85,11 +86,13 @@ def __init__(self, vars=None, path_length=2.,
             The model
         **kwargs : passed to BaseHMC
         """
+        kwargs.setdefault('step_rand', unif)
+        kwargs.setdefault('target_accept', 0.65)
         super().__init__(vars, **kwargs)
         self.path_length = path_length
 
     def _hamiltonian_step(self, start, p0, step_size):
-        path_length = np.random.rand() * self.path_length
```
Comment (junpenglao) [inline pymc3/step_methods/hmc/hmc.py:92]: Do we really want to remove this? If so `n_steps = max(1, int(path_length / step_size))` is constant and we dont need to recomputed it in each step any more.

Comment (ColCarroll) [reply pymc3/step_methods/hmc/hmc.py:92]: https://github.com/pymc-devs/pymc3/blob/master/pymc3/step_methods/hmc/base_hmc.py#L151 takes care of that (using `step_rand`)

Comment (junpenglao) [reply pymc3/step_methods/hmc/hmc.py:92]: Oh OK, good catch of the bug.

File: pymc3/step_methods/hmc/hmc.py [hunk 2]
```diff
@@ -85,11 +86,13 @@ def __init__(self, vars=None, path_length=2.,
...
+        path_length = self.path_length
```
Comment (junpenglao) [inline pymc3/step_methods/hmc/hmc.py]:
```suggestion
        n_steps = max(1, int(self.path_length / step_size))
```
?

Comment (ColCarroll) [reply pymc3/step_methods/hmc/hmc.py]: good idea -- made the change

**Comment by junpenglao** [comment]
@ColCarroll could you check the benchmark at some point - curious to see whether it gives better ESS.

**Pull request merged**